### ISSUE TITLE
Rule proposal: `prefer-string-repeat`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1708]
### Description

There are legacy way to repeat string, we should replace them with `.repeat()`, but this proposal is not about that. I want enforce `.repeat()` for unreadable string literals.

### Fail

```js
const string = '    '; // I can see how many are there
```

```js

const string = 'unicorn unicorn unicorn unicorn';
```

### Pass

```js
const string = ' '.repeat(4);
```

```js
const string = 'unicorn '.repeat(4).trim();  // This?
const string = Array.from({length: 4}).fill('unicorn').join(' '); // Or this?
const string = Array.from({length: 4}, () => 'unicorn').join(' '); // Or this?
```

**Comment by sindresorhus** [comment]
> const string = 'unicorn '.repeat(4).trim();  // This?
> const string = Array.from({length: 4}).fill('unicorn').join(' '); // Or this?
> const string = Array.from({length: 4}, () => 'unicorn').join(' ');

The third one is slightly more efficient, so I think we should recommend that one.

**Comment by silverwind** [comment]
The minimum number of repetions should be configurable so one can avoid it triggering on still-readable stuff like string `'aa'`;

**Comment by sindresorhus** [comment]
Let's go with `3` for now. We can increase it to `4` if `3` is annoying in practice.

**Comment by sindresorhus** [comment]
Accepted

**Comment by fisker** [comment]
`no-repeated-string`? Not all cases should recommend `String#repeat()`.

**Comment by silverwind** [comment]
> `const string = 'unicorn unicorn unicorn unicorn';`

Technically this isn't strictly a completely repeated string. Would it detect 3 or 4 repetitions here?

**Comment by fisker** [comment]
4, my idea is to consider `unicorn unicorn unicorn ` as 3 repetitions, then compare string after it, we can result `unicorn` will be the repeated string(4), space will be the seperator.

But I'm not sure if this a good solution, for example nobody want change `idea idea idea id` to  `Array.from({length: 3}, () => 'id').join('ea ');`.

**Comment by silverwind** [comment]
Generally, I think I'm leaning towards a higher default, 4 or even more.

**Comment by fisker** [comment]
But `   ` already unreadable. We need a good algorithm here.

**Comment by fregante** [comment]
> But `"   "` is already unreadable

That's unreadable because it's made invisible of characters; `"aaaa"` is perfectly readable as humans can [subitize](https://en.wikipedia.org/wiki/Subitizing) a small number of items.

**Comment by fregante** [comment]
I don’t think this rule is useful other than for repeating spaces. How often do you have repeating characters in a string? I'd probably make a whitespace-specific rule instead, which can cover tabs

```js
'  '        -> ' '.repeat(2)
'	'       -> '\t'
'		'   -> '\t'.repeat(2)
' '         -> '\u2003' // EM space
```

More spaces here: https://codepen.io/rubikzube/pen/kKYZwW

**Comment by fisker** [comment]
Make sense to me 👍 .
